adcpipeline/pipeline.py

```diff
@@ -72,9 +74,13 @@ def method_list(self) -> List[Callable]:
         """
         return self._method_list
 
-    def __init__(self, df: DataFrame, method_settings: List) -> None:
+    def __init__(self, df: DataFrame, method_settings: List, filename: str = None) -> None:
```

Comment:
Small tiny request by the way, could you add `df: Optional[DataFrame]`? I can see now that you really need the df for the caching of course, but my PyCharm doesn't like the `None` option ;) We could also switch the arguments around, and set the `df` argument by default on `None`.

Reply:
Done